db/blip_handler.go

```diff
@@ -551,7 +551,7 @@ func (bh *blipHandler) sendBatchOfChanges(sender *blip.Sender, changeArray [][]i
 		bh.replicationStats.SendChangesCount.Add(int64(len(changeArray)))
 		// Spawn a goroutine to await the client's response:
 		go func(bh *blipHandler, sender *blip.Sender, response *blip.Message, changeArray [][]interface{}, sendTime time.Time, dbCollection *DatabaseCollectionWithUser) {
-			if err := bh.handleChangesResponse(sender, response, changeArray, sendTime, dbCollection); err != nil {
+			if err := bh.handleChangesResponse(sender, response, changeArray, sendTime, dbCollection, bh.collectionIdx); err != nil {
```

Comment:
It makes me wary, but I'm sure that there's a good reason to pass the collection index here.

My understanding was that we wanted a copy of the state of the ~database~ collection before we start replication, so at some point (not in this review?) we're going to be able to access this again. I might have done this anyway, which is why you removed `getIndexForDB`. In that case, I wonder if it makes sense to pass two collections?

I don't object to this change, just trying to understand.

Reply:
Passing collectionIdx directly avoids having to iterate over the set of collections to find the index matching the collection name via `getIndexForDB` - which seemed way too inefficient to do for every revision.

We know the set of collections on a replication won't change so this seems safe to me?